File: lp2/PokemonV-Java/src/lp2/Pokemon.java
package lp2;

enum Type {
    Fire, Water, Earth, Wind
};

public class Pokemon {

    public int id;
    public float attack_pts;
    public float defense_pts;
    public float life;
    public String name;
    public Type type;

    public Pokemon(int id, int attack_pts, int defense_pts, int life, String name, Type type) {
        this.id = id;
        this.attack_pts = attack_pts;
        this.defense_pts = defense_pts;
        this.life = life;
        this.name = name;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public float getAttack_pts() {
        return attack_pts;
    }

    public void setAttack_pts(int attack_pts) {
        this.attack_pts = attack_pts;
    }

    public float getDefense_pts() {
        return defense_pts;
    }

    public void setDefense_pts(int defense_pts) {
        this.defense_pts = defense_pts;
    }

    public float getLife() {
        return life;
    }

    public void setLife(int life) {
        this.life = life;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public void attack(Pokemon a) {
        
    }

    public void takeDamage(float attack_points) {
        
    }
}

File: lp2/PokemonV-Server/Entidades/Pokemon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entidades
{
    public enum TypeofPokemon { Fire, Water, Earth, Wind };
    public class Pokemon
    {
        public Pokemon(string name, int life, int defense_pts, TypeofPokemon type, int attack1id, int attack2id, int deleted )
        {   
            this.defense_pts = defense_pts;
            this.life = life;
            this.name = name;
            this.type = type;
            this.attack1_id = attack1id;
            this.attack2_id = attack2id;
            this.deleted = deleted;
        }

        public void attack(Pokemon pokemon)
        {

        }
        public void takeDamage(float attack_pts)
        {

        }



        private int id;
        public int Id
        {
            get { return id; }
            set { id = value; }
        }
        
       
        private int defense_pts;
        public int Defense_pts
        {
            get { return defense_pts; }
            set { defense_pts = value; }
        }
        private int life;
        public int Life
        {
            get { return life; }
            set { life = value; }
        }
        private string name;
        public string Name
        {
            get { return name; }
            set { name = value; }
        }
        private TypeofPokemon type;
        public TypeofPokemon Type
        {
            get { return type; }
            set { type = value; }
        }

        public int Deleted { get => deleted; set => deleted = value; }
        public int Attack1_id { get => attack1_id; set => attack1_id = value; }
        public int Attack2_id { get => attack2_id; set => attack2_id = value; }

        int attack1_id;
        int attack2_id;
        private int deleted;
    }
}
